app/Http/Controllers/API/v1/TransactionController.php

```diff
@@ -284,9 +284,8 @@ public function store(Request $request): JsonResponse
                     // schedule next task.
                     $next_date = get_next_transaction_schedule_date($recurring_transaction);
                     RecurrentTransactionJob::dispatch(
-                        id: (int) $recurring_transaction->id,
-                        recurrence_period: $recurring_transaction->recurrence_period,
-                        recurrence_interval: $recurring_transaction->recurrence_interval,
```

Comment:
@iMercyvlogs the `recurrence_period` and `recurrence_interval` were sent to the job to determine if the rule is still the same by the time the job runs. It could be possible for a user to change the recurrence period or frequency by the time the next job runs. That is why we check if the recurrence period and interval are the same before running the job.

Please test again and ensure the next job (that was already scheduled) does not run if the interval and period changes and let me know if it your new changes handle this scenario.